plugins/modules/wait_for_task.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2023, Julien Godin <julien.godin@camptocamp.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: wait_for_task
+version_added: 3.13.0
+short_description: Wait for a task
+description:
+  - Wait for a task to finish
+author:
+  - "Julien Godin (@JGodin-C2C)"
+options:
+  task:
+    description:
+      - Task to wait for. Typical output of the resource_info module
+    required: true
+    type: dict
+  timeout:
+    description:
+      - How much time the task should take to be finished
+    required: false
+    type: int
+    default: 60
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+'''
+
+
+EXAMPLES = '''
+- name: wait for a task to finish in at least 10 seconds
+  theforeman.foreman.wait_for_task:
+    server_url:  "https://foreman.example.com"
+    password: changeme
+    username: admin
+    task: "{{ item }}"
+    timeout: 60
+  loop: "{{ tasks.resources }}"
+
+'''
+
+RETURN = '''
+entity:
+  description: Designated task is finished
+  returned: success
+  type: dict
+  contains:
+    task:
+        description: The finished task
+        type: dict
+'''
+
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import ForemanAnsibleModule
+
+
+class ForemanWaitForTask(ForemanAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanWaitForTask(
+        argument_spec=dict(
+            task=dict(type="dict", required=True),
+            timeout=dict(type="int", required=False, default=60),
+        )
+    )
+    module.task_timeout = module.foreman_params["timeout"]
+    with module.api_connection():
+        module.wait_for_task(module.foreman_params["task"])
```

Comment:
```suggestion
        task = module.wait_for_task(module.foreman_params["task"])
        module.exit_json(task=task)
```

that way we return the task details so users can take further actions based on the finished task result

Reply:
Ok, but i think it is smarter to return the same type as the one passed to this module. Waiting for the other comment to be resolved.

Reply:
We can also do `module.exit_json(task=task, task_id=task['id'])` to have both directly available?

My trail of thought was that:
- users might want to have the additional data here, because later steps might depend on details of the finished task
- users might want to start waiting based on id only (as that's really the only data we need) because they get the ID by some other means than `resource_info`

Reply:
Is returning both task AND task["id"] useful thought ? I mean, the ID is already in the task that just finished. 

I think the best return value would be the task, as you mentionned, users may need additional info.

Reply:
So you could use `result.task.id` for another call to this module, right?
In a way, this makes this module like a `task_info` module that also waits on the task first. (Not saying this is bad.)

Reply:
Done, it now return both values.